pyproject.toml

```
@@ -35,12 +56,115 @@ distance-dirty = "{next_version}.dev{distance}+d{build_date:%Y%m%d%H%M}"
 [tool.versioningit.write]
 file = "src/quicknxs/_version.py"
 
-[tool.setuptools.packages.find]
-where = ["src"]
-exclude = ["tests*", "scripts*", "docs*", "notebooks"]
+###################
+### Pixi config ###
+###################
+
+[tool.pixi.workspace]
+name = "quicknxs"
+platforms = ["linux-64"]
+channels = [
+    "conda-forge",
+    "neutrons",
+    "oncat",
+    "mantid-ornl",
+    "mantid",
+    "https://prefix.dev/pixi-build-backends",
+]
+# Required until the build feature becomes stable
+preview = ["pixi-build"]
+
+[tool.pixi.package]
+name = "quicknxs"
+version = "0.0.0"
+
+[tool.pixi.package.build]
+backend = { name = "pixi-build-python", version = "0.1.*" }
 
-[tool.setuptools.package-data]
-"*" = ["*.yml", "*.yaml", "*.ini", "*.gx", "*.json", "*.ui"]
+[tool.pixi.package.host-dependencies]
+hatchling = "*"
+versioningit = "*"
+
+[tool.pixi.dependencies]
+mr_reduction = "==2.5.0"
+
+[tool.pixi.pypi-dependencies]
+quicknxs = { path = ".", editable = true }
+
+# Environments and features
+
+[tool.pixi.environments]
+default = { features = [
+    "dev",
+    "docs",
+    "test",
+    "build",
+], solve-group = "default" }
+docs = { features = ["dev", "docs"], no-default-feature = true }
+build = { features = ["build"], no-default-feature = true }
+
+[tool.pixi.feature.build.dependencies]
+anaconda-client = "*"
+check-wheel-contents = "*"
+versioningit = "*"
+
+[tool.pixi.feature.build.pypi-dependencies]
+toml-cli = "*"
+
+[tool.pixi.feature.dev.dependencies]
+pip = "*"
+pip-audit = "*"
+pre-commit = "*"
+ruff = "*"
+versioningit = "*"
+
+[tool.pixi.feature.docs.dependencies]
+sphinx = "*"
+sphinx_rtd_theme = "*"
+
+[tool.pixi.feature.docs.pypi-dependencies]
+sphinx-qt-documentation = "*"
+
+[tool.pixi.feature.test.dependencies]
+codecov = "*"
+pytest = "*"
+pytest-qt = "*"
+pytest-cov = "*"
+pytest-mock = "*"
+pytest-xvfb = "*"
+
+[tool.pixi.tasks]
+# Documentation
+build-docs = { cmd = 'sphinx-build -b html docs docs/_build/html', description = "Build documentation" }
+test-docs = { cmd = "sphinx-build -M doctest docs docs/_build/html", description = "Test building the documentation" }
+# Testing
+test = { description = "Run the test suite", cmd = "pytest" }
+# Packaging
+conda-build-command = { cmd = "pixi build", description = "Wrapper for building the conda package - used by `conda-build`" }
+conda-build = { description = "Build the conda package", depends-on = [
+    "sync-version",
+    "conda-build-command",
+    "reset-version",
+] }
+conda-publish = { cmd = "anaconda upload *.conda", description = "Publish the .conda package to anaconda.org", depends-on = [
+    "conda-build",
+] }
+# Misc
+audit-deps = { cmd = "pip-audit --local -s osv", description = "Audit the package dependencies for vulnerabilities" }
+clean = { cmd = 'rm -rf .pytest_cache .ruff_cache **/*.egg-info **/dist **/__pycache__ **_version.py', description = "Clean up various caches and build artifacts" }
```

This one generated a "pattern syntax error", maybe like this?

```suggestion
clean = { cmd = 'rm -rf .pytest_cache .ruff_cache **/*.egg-info **/dist **/__pycache__ **/_version.py', description = "Clean up various caches and build artifacts" }
```